etl/steps/data/garden/health/2024-04-12/polio_free_countries.py

```diff
@@ -0,0 +1,116 @@
+    tb = define_polio_free(tb, latest_year=LATEST_YEAR)
+
+    tb = add_polio_region_certification(tb, tb_region_status, who_regions, ds_regions)
+    # Set an index and sort.
+    tb = tb.format()
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset.
+    ds_garden = create_dataset(dest_dir, tables=[tb], check_variables_metadata=True)
+    ds_garden.save()
+
+
+def add_polio_region_certification(
+    tb: Table, tb_region_status: Table, who_regions: Table, ds_regions: Dataset
+) -> Table:
+    # Append "(WHO)" suffix to the "who_region" to match the region names in the who_regions table
+    tb_region_status["who_region"] = tb_region_status["who_region"].astype(str) + " (WHO)"
+
+    # Correct mapping of regions to status updates by ensuring 'region' matches the modified 'who_region' entries
+    for region in who_regions["name"]:
+        # Generate country list for the current region
+        country_list = list_members_of_region(region=region, ds_regions=ds_regions)
+        if not country_list:
+            raise ValueError(f"No countries found for region {region}")
+
+        # Find the year of certification for the current region
+        year_certified = tb_region_status.loc[tb_region_status["who_region"] == region, "year_certified_polio_free"]
+
+        # Check if there is a valid year of certification
+        if not year_certified.empty and year_certified.notna().all():
+            year_certified_int = int(year_certified.iloc[0])
+            # Set the status for all relevant countries and years
+            tb.loc[
+                (tb["country"].isin(country_list)) & (tb["year"] >= year_certified_int), "status"
+            ] = "WHO Region certified polio-free"
+
+    return tb
+
+
+def define_polio_free(tb: Table, latest_year: int) -> Table:
+    """Define the polio free countries table."""
+    tb = tb.copy()
+    # Clean the data
+    tb["year"] = tb["year"].astype(str)
+
+    # Drop countries with missing values explicitly copying to avoid setting on a slice warning
+    tb = tb[tb["year"] != "data not available"].copy()
+
+    # Change 'pre 1985' to 1984 and 'ongoing' to LATEST_YEAR + 1
+    tb.loc[tb["year"] == "pre 1985", "year"] = "1984"
+    tb.loc[tb["year"] == "ongoing", "year"] = str(latest_year + 1)
+
+    tb["year"] = tb["year"].astype(int).copy()
+    # Rename year to latest year
+    tb = tb.rename(columns={"year": "latest_year_wild_polio_case"})
+    tb["year"] = latest_year
+    # Create a product of all countries and all years from 1910 to LATEST_YEAR
+    tb_prod = Table(product(tb["country"].unique(), range(1910, latest_year + 1)), columns=["country", "year"])
+    tb_prod = tb_prod.copy_metadata(from_table=tb)
+
+    # Define polio status based on the year comparison
+    tb_prod["status"] = tb_prod.apply(
+        lambda row: "Endemic"
+        if row["year"] < tb[tb["country"] == row["country"]]["latest_year_wild_polio_case"].min()
+        else "Polio-free (not certified)",
```

Comment:
The wording of "not certified" can be confusing, because that only depends on the region status. How about `Polio-free in a not certified WHO region`, or something similar?

Reply:
Will check with Saloni and see what she says